test/tools/test_component_tool.py

```diff
@@ -122,6 +122,26 @@ def run(self, documents: List[Document], top_k: int = 5) -> Dict[str, str]:
         return {"concatenated": "\n".join(doc.content for doc in documents[:top_k])}
 
 
+@component
+class ComplexTypeProcessor:
+    """A component that processes complex types."""
+
+    @component.output_types(result=str)
+    def run(
+        self, meta: Union[Dict[str, Any], List[Dict[str, Any]]] = None, extraction_kwargs: Dict[str, Any] = None
```

Comment:
Could we make one of these optional like in the original `run` method of `HTMLToDocument`? It'd be good to check that your solution works with `Optional` type as well. 
```suggestion
        self, meta: Union[Dict[str, Any], List[Dict[str, Any]]] = None, extraction_kwargs: Optional[Dict[str, Any]] = None
```

Reply:
Done! Since we're now handling dictionaries properly in `_create_property_schema`, I've removed it from the basic type mapping.